docs/architecture/views.md

```diff
@@ -0,0 +1,169 @@
+
+[`memo()`](../api/memo.md) is designed to be used with other functions that produce VNodes, so it doesn't really describe a view by itself.
+
+```js
+const view = (state) => memo(scenicView, state.vacationSpot)
+```
+<!-- Just a play-on-words between how "view" is used in Hyperapp and everyday language. -->
+
+---
+
+## Components
+
+Views are naturally composable so they can be as simple or complicated as you need. Simpler apps probably just need a single view but in more complicated apps there could be plenty of subviews.
+
+A **component** in Hyperapp can either be a subview or some other function that generates VNodes.
+
+You would typically make components for widgets that provide the building block elements of your app's UI. Components for larger UI segments such as dashboards or pages would make use of these widgets.
+
+In the following example, `coinsDisplay` is a component in the form of a subview while `questionBlock` is a component in the form of some function that returns a VNode. Notice the former cares directly about the state while the latter doesn't:
+
+```js
+const coinsDisplay = (state) =>
+  h("div", { class: "coins-display" }, text(state.coins))
+
+const questionBlock = (opened) =>
+  opened
+    ? h("button", { class: "question-block opened" }, text("?"))
+    : h(
+        "button",
+        {
+          class: "question-block",
+          onclick: [
+            HitBlockFromBottom,
+            { revealItem: "beanstalk" },
+          ],
+        },
+        text("?")
+      )
+
+const level = (state) =>
+  h("div", { class: "level" }, [
+    coinsDisplay(state),
+    questionBlock(state.onlyQuestionBlockOpened),
+  ])
+```
+<!-- In the videogame "Super Mario Bros." coins are important for earning extra lives and the question blocks often contain useful contents. -->
+
+### Components Returning Multiple VNodes
+
+Components are allowed to return an array of VNodes. However, to make use of such components you'll need to spread their result.
+
+```js
+const finishingMoveOptions = () => [
+  h("button", { onclick: FinishHim }, text("Fatality")),
+  h("button", { onclick: FinishHimAsAnAnimal }, text("Animality")),
+  h("button", { onclick: TurnHimIntoABaby }, text("Babality")),
+  h("button", { onclick: BefriendHim }, text("Friendship")),
+]
+
+const view = () => h("div", {}, ...finishingMoveOptions())
+```
+<!-- In the "Mortal Kombat" videogame series there are multiple ways to finish off your opponent. The opportunity to do so occurs at the end of a match once the match announcer exclaims "Finish Him!" -->
+
+---
+
+## Using Views
+
+### Top-Level View
+
+Every Hyperapp application has a base view that encompasses all others. This is the **top-level view** that's defined by the [`view:`](../api/app.md#view) property when using [`app()`](../api/app.md). Hyperapp automatically calls this view and gives it the current state when the state is initially set or anytime it's updated.
```

Comment:
Using the word "application" here could be potentially confusing since it is not unreasonable to architect a single web-app using multiple instances of `app()` calls.  

Maybe I would rather say: "Every call to `app()` sets up a hyperapp instance which manages a single particular node in the DOM, as defined by the `view` property. This is the top-level view of the app."

Reply:
I'll look into using "instance" in place of "application" since the latter is used in multiple places I believe.